File: 1642-water-bottles/water-bottles.cpp
class Solution {
public:
    int numWaterBottles(int numBottles, int numExchange) {
        // 07.07.24 POTD
        int ans = numBottles;

        while(numBottles >= numExchange){
            numBottles += 1 - numExchange;
            ans++;
        }

        return ans;
    }
};

File: 1642-water-bottles/water-bottles.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        # 01.10.25 POTD
        ans = 0
        emp = 0
        nb, ne = numBottles, numExchange

        while nb > 0:
            ans += nb
            nb2 = (emp + nb) // ne
            emp = (emp + nb) % ne
            nb = nb2
            print(nb, emp)
        
        return ans